streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableRepartitionMap.java

```diff
@@ -66,32 +65,25 @@ public void enableSendingOldValues() {
         throw new IllegalStateException("KTableRepartitionMap should always require sending old values.");
     }
 
-    private KeyValue<K1, V1> computeValue(K key, V value) {
-        KeyValue<K1, V1> newValue = null;
-
-        if (key != null || value != null)
-            newValue = mapper.apply(key, value);
-
-        return newValue;
-    }
-
     private class KTableMapProcessor extends AbstractProcessor<K, Change<V>> {
 
         /**
          * @throws StreamsException if key is null
          */
         @Override
         public void process(K key, Change<V> change) {
-            KeyValue<K1, V1> newPair = computeValue(key, change.newValue);
-
-            // the selected repartition key should never be null
-            if (newPair.key == null)
-                throw new StreamsException("Record key for KTable repartition operator should not be null.");
+            // the original key should never be null
+            if (key == null)
+                throw new StreamsException("Record key for the grouping KTable should not be null.");
 
-            context().forward(newPair.key, new Change<>(newPair.value, null));
+            KeyValue<K1, V1> newPair = mapper.apply(key, change.newValue);
+            KeyValue<K1, V1> oldPair = mapper.apply(key, change.oldValue);
 
-            if (change.oldValue != null) {
-                KeyValue<K1, V1> oldPair = computeValue(key, change.oldValue);
+            // if the selected repartition key or value is null, skip
+            if (newPair.key != null && newPair.value != null) {
```

Comment:
Should we check `newPair != null` before checking the key and value for nulls? A user might want to have their `mapper` return `null` in the case they want the value to be skipped.